src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -16068,18 +16068,21 @@ public static Vector512<T> Invoke(Vector512<T> x)
 
             public static Vector128<float> Invoke(Vector128<float> x)
             {
-                Vector128<uint> uxMasked = Vector128.Abs(x).AsUInt32();
-                if (Vector128.GreaterThanAny(uxMasked, Vector128.Create(MaxVectorizedValue)))
+                Vector128<float> uxMasked = Vector128.Abs(x);
+                if (Vector128.GreaterThanAny(uxMasked.AsUInt32(), Vector128.Create(MaxVectorizedValue)))
                 {
                     return ApplyScalar<CosOperatorSingle>(x);
                 }
 
-                Vector128<float> r = uxMasked.AsSingle();
                 Vector128<float> almHuge = Vector128.Create(AlmHuge);
-                Vector128<float> dn = ((r + Vector128.Create(float.Pi / 2)) * Vector128.Create(1 / float.Pi)) + almHuge;
+                Vector128<float> dn = MultiplyAddEstimateOperator<float>.Invoke(uxMasked + Vector128.Create(float.Pi / 2), Vector128.Create(1 / float.Pi), almHuge);
                 Vector128<uint> odd = dn.AsUInt32() << 31;
                 dn = dn - almHuge - Vector128.Create(0.5f);
-                Vector128<float> f = r + (dn * Vector128.Create(-float.Pi)) + (dn * Vector128.Create(Pi_Tail1)) + (dn * Vector128.Create(Pi_Tail2));
+                Vector128<float> f =
+                    MultiplyAddEstimateOperator<float>.Invoke(dn, Vector128.Create(-float.Pi),
+                        MultiplyAddEstimateOperator<float>.Invoke(dn, Vector128.Create(Pi_Tail1),
+                            MultiplyAddEstimateOperator<float>.Invoke(dn, Vector128.Create(Pi_Tail2),
+                                uxMasked)));
```

Comment:
This has changed how the results compound together from:
```csharp
var f = r;
f += (dn * Vector128.Create(Pi_Tail2));
f += (dn * Vector128.Create(Pi_Tail1));
f += (dn * Vector128.Create(-float.Pi);
```

It needs to maint the same ordering, and thus you need to inverse the multiplications (inner-most calls happen first, not last as compared to expressions which are left to right based on precedence)

to:
```csharp
var f = r;
f += (dn * Vector128.Create(-float.Pi);
f += (dn * Vector128.Create(Pi_Tail1));
f += (dn * Vector128.Create(Pi_Tail2));
```

Reply:
I think it might even be helpful to break it apart like that for readability.
```csharp
Vector128<float> f = r;
f = MultiplyAddEstimate(dn, Vector128.Create(-float.Pi), f);
f = MultiplyAddEstimate(dn, Vector128.Create(Pi_Tail1), f);
f = MultiplyAddEstimate(dn, Vector128.Create(Pi_Tail2), f);
```